src/main/java/ch/epfl/biop/operetta/OperettaManager.java

```diff
@@ -839,6 +895,30 @@ public void process(List<Well> wells, List<Integer> fields, Roi region) {
         double percentageCompleteness;
 
         try {
+            // create the companion generator
+            CompanionFileGenerator companionFileGenerator = new CompanionFileGenerator();
+
+            // get the global metadata, those which are displayed under the OriginalMetadata tab on OMERO
+            Map<String, String> globalMetadataMap = this.main_reader.getGlobalMetadata()
+                    .entrySet()
+                    .stream()
+                    .collect(Collectors.toMap(Map.Entry::getKey, e->e.getValue().toString()));
+
+            // create the plate object
+            Plate plate = getPlate();
+            PlateCompanion plateCompanion = new PlateCompanion.Builder()
+                    .setName(getPlateName())
+                    .setNRows(plate.getRows().getValue())
```

Comment:
Are you sure getValue is the correct thing ? Be aware that people can acquire and / or export only certain wells